internal/controller/argocd.go

```diff
@@ -150,6 +150,31 @@ func (clctrl *ClusterController) InitializeArgoCD() error {
 	return nil
 }
 
+
+func RestartDeployment(ctx context.Context,clientset *kubernetes.Clientset,namespace string,deployment_name string) error {
```

Comment:
It would be good to have a unit test for `RestartDeployment` so if someone like me were to go modify the function, I don't break your work.

Reply:
We do need to check that our CI/CD pipelines actually run tests. I can't see anything in the `.github/workflows` folder with `go test` (or similar)

I would suggest creating the test as part of this PR and then doing a separate ticket/PR to implement that

Reply:
To avoid scope creeping @jokestax (and because he's probably already past biz hours) I'm happy to take that part of allowing unit tests to run.

Reply:
Added unit test and rebased